### PR TITLE
Refactoring of AWS module

### PR DESCRIPTION
**Pull request by vg-svitla**

### CODE REVIEW

File: AWS/aws_helpers/sqs_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+                logger.info(f"Received {len(result)} messages from sqs queue {self._configuration.queue_name}")
+
+                yield result
+            finally:
+                # We should delete messages from queue after releasing context manager if it is configured
+                if delete_consumed_messages and response.get("Messages", []):
+                    logger.info("Deleting consumed messages from sqs")
+                    for message in response.get("Messages", []):
+                        await sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=message["ReceiptHandle"])
```
Comment (squioc) [inline AWS/aws_helpers/sqs_wrapper.py:131]: Does the async sqs client not allow messages to be deleted in batch? Like [this code](https://github.com/SEKOIA-IO/automation-library/blob/main/AWS/aws/sqs/trigger_sqs_messages.py#L64-L67)

File: AWS/connectors/__init__.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        loop = asyncio.get_event_loop()
+
+        previous_processing_end = None
+        try:
+            while True:
+                processing_start = time.time()
+                if previous_processing_end is not None:
+                    EVENTS_LAG.labels(intake_key=self.configuration.intake_key).observe(
+                        processing_start - previous_processing_end
+                    )
```
Comment (squioc) [inline AWS/connectors/__init__.py:76]: The SQS [ReceiveMessage](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html) action allow you to get the `SentTimestamp` attribute name. I wonder if using this attribute to calculate the delay is more appropriate. Could you take a look?
Thanks

File: AWS/connectors/trigger_sqs_messages.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+from sekoia_automation.connector import DefaultConnectorConfiguration
+
+from aws_helpers.sqs_wrapper import SqsConfiguration, SqsWrapper
+from connectors import AbstractAwsConnector
+
+
+class AwsSqsMessagesTriggerConfiguration(DefaultConnectorConfiguration):
+    """Contains configuration for AwsSqsMessagesTrigger."""
+
+    frequency: int = 60
```
Comment (squioc) [inline AWS/connectors/trigger_sqs_messages.py:14]: This parameter seems to not be used in the connector? Is useless or a `asyncio.sleep` is missing?

**Review by squioc — COMMENTED** [review]
Thanks for the job.

I got some questions.

File: AWS/connectors/trigger_sqs_messages.py (hunk 3 above)
Comment (vg-svitla) [reply AWS/connectors/trigger_sqs_messages.py:14]: Yep! Thank you! Fixed

File: AWS/aws_helpers/sqs_wrapper.py (hunk 1 above)
Comment (vg-svitla) [reply AWS/aws_helpers/sqs_wrapper.py:131]: As per docs from here https://github.com/aio-libs/aiobotocore#supported-aws-services `sqs` wrapper is only partially supported.
I was not able to find any tests that  proves that delete batch operation works as expected so I tried myself.
I was not able to use it, as I faced generic `Exception` with strange message.
I can investigate it more, or we can try to use custom implementation

File: AWS/connectors/__init__.py (hunk 2 above)
Comment (vg-svitla) [reply AWS/connectors/__init__.py:76]: If I'm correct the main goal of `EVENTS_LAG` is to check delay between runs in connector.
As per official docs `SentTimestamp` is the time when the message was sent to the queue.
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html
Lets discuss in DM

**Review by squioc — APPROVED** [review]

**Pull request merged**